src/rules/no_unused_vars.rs

```diff
@@ -83,6 +84,10 @@ struct Collector {
   /// Type of this should be hashset, but we don't have a way to
   /// restore hashset after handling bindings
   cur_defining: Vec<Id>,
+
+  /// Whether the code has JSX tag syntax or not.
+  /// If it does, imported `React` has to be treated as implicitly "used".
```

Comment:
The char that is used for this can be changed using the tsconfig file. When using `preact` the implicitly used vars are `h` and `Fragment`.

Reply:
Oh I missed that point, thanks.
I'm not so familiar with this sort of thing, but from looking at the following documents, is it sufficient if just all of `React`, `h` and `Fragment` are marked as "used"? Or can developers choose arbitrary names they want to use as JSX factory function, even in the current version of TypeScript?

https://preactjs.com/guide/v10/typescript/
https://www.typescriptlang.org/docs/handbook/jsx.html

Reply:
Yes, they can choose arbitrary names. There are two configuration options for this in the tsconfig. `jsxFactory` and `jsxFragmentFactory`. The fragment factory name is used for JSX fragments (`<></>`), while the regular JSX factory is used for everything else (`<div></div>`, `<Foo />`). I suggest you have the implementor specify the `jsxFactory` and `jsxFragmentFactory` when configuring the linter.

Reply:
That's unfortunate, things get complicated😣 but thanks for the clarification.

>  I suggest you have the implementor specify the jsxFactory and jsxFragmentFactory when configuring the linter.

That would be ideal, but however at the moment there's no capability of configuration, and moreover, I don't think deno_lint should be configurable in the way the behavior of rules can be changeable. Instead, I'd prefer to put a restriction on the implementors so that they _cannot_ choose arbitrary names for `jsxFactory` or `jsxFragmentFactory`. (Not quite sure if it's feasible though)

Reply:
The implementer can already get these values from the tsconfig, so it would not be "extra" configuration. @bartlomieju WDYT?

Reply:
Maybe we just have to disable this rule on JSX files.

Reply:
> Maybe we just have to disable this rule on JSX files.

It sounds reasonable to me. False negative is more acceptable than false positive.

Reply:
I'm fine with both approaches. Right now `deno lint` does not take configuration flag, but we could potentially add it to sniff out values from tsconfig.

Reply:
Just fixed so it will skip linting `.jsx` and `.tsx` files. Please take a look again.

Reply:
Looks ok to me